### DISCUSSION TITLE
Prioritize longer extensions in `file-types`

### DISCUSSION THREAD

**Discussion by nitsky** [helix-editor/helix#9289] [General]
Hi, I want to create a language that uses the file extension `.foo.ts`, a special flavor of typescript that uses its own language server. However, I find that setting the `file-types` key to `["foo.ts"]` doesn't work. The file still gets recognized as typescript. It appears that helix does not do prioritization based on length the same way that vscode and other editors do. I would like to propose that helix find all matching file-types extensions, and then choose the longest.

**Comment by pascalkuthe** [comment]
the issue has nothing to do with length her but rather that we (or rather the rust standard library) only considers the text after the last period as the file extension. If you want to recognize such a file extension you have to use a suffix (will be covered by globs in the future see #8006)
